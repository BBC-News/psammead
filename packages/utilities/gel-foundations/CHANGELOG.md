# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.4   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.1   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.0   | [PR#221](https://github.com/BBC-News/psammead/pull/221) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |

