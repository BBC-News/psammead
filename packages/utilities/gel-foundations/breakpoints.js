module.exports = require('./dist/breakpoints'); // eslint-disable-line import/no-unresolved
