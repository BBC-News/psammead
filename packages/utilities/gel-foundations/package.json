{
  "name": "drew-testing-123",
  "version": "1.0.6",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "description": "A range of string constants for use in CSS, intended to help implement BBC GEL-compliant webpages and components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/utilities/gel-foundations"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "files": [
    "dist/"
  ],
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "gel",
    "breakpoints",
    "grid",
    "spacing",
    "typography"
  ],
  "devDependencies": {
    "@bbc/psammead-test-helpers": "^1.0.1"
  }
}
