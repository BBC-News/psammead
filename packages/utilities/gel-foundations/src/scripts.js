export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
