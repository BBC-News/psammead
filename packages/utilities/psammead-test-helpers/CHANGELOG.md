# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.3.0   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Adds a helper method for testing package export values and uses it to test its own exports. |
| 0.2.1   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
