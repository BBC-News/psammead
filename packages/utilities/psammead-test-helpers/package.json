{
  "name": "@bbc/psammead-test-helpers",
  "version": "0.3.0",
  "main": "dist/index.js",
  "description": "A collection of helper methods for implementing Jest snapshot tests for styled-components, required by many Psammead components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/BBC-News/psammead/tree/latest/packages/utilities/psammead-test-helpers"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/BBC-News/psammead/blob/latest/packages/utilities/psammead-test-helpers/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "test",
    "helpers",
    "jest",
    "snapshot"
  ],
  "dependencies": {
    "jest-styled-components": "^6.3.1",
    "react-test-renderer": "^16.6.3"
  }
}
