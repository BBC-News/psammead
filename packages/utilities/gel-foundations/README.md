# gel-foundations &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/bbc/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/gel-foundations.svg)](https://www.npmjs.com/package/@bbc/gel-foundations) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md)

This package provides a range of string constants for use in CSS, intended to help implement [BBC GEL-compliant](https://www.bbc.co.uk/gel/articles/what-is-gel) webpages and components.

## Exports

`/breakpoints` - GEL breakpoints, as well as typography breakpoints. These use the GEL grid sizes which can be found in the [GEL Grid guidelines](https://www.bbc.co.uk/gel/guidelines/grid#grid-sizes).  
`/spacings` - GEL spacings and GEL Grid margins and gutters. These use the GEL grid spacing sizes which can be found in the [GEL Grid guidelines](https://www.bbc.co.uk/gel/guidelines/grid#spacing-layout).  
`/typography` - GEL typography. These are based on the gel typography standard which can be found in the [GEL guidelines](https://www.bbc.co.uk/gel/guidelines/typography). [More details on how these sizes were implemented for this package are available here.](./typography_sizes_web.md)

## Installation

```jsx
npm install @bbc/gel-foundations --save
```

## Usage

```jsx
import { GEL_GROUP_3_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';

import { GEL_GUTTER_ABOVE_600PX } from '@bbc/gel-foundations/spacings';

import { GEL_BREVIER } from '@bbc/gel-foundations/typography';
```

These values can then be used directly within CSS declarations in code:

```jsx
import { css } from 'styled-components';

const SomeStyledComponent = css`
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    grid-gap: ${GEL_GUTTER_ABOVE_600PX};
    ${GEL_BREVIER};
  }
`;
```

To allow the typography to be fully accessible and responsive, please note that you should apply a default font-size to the document root (e.g. `html { font-size: 100% }`).

Our typography uses `em` for font-size and `rem` for line-height. `em` allows modularity of components: you can change the component font-size by changing the font-size of its container. `rem` is relative to the document root, so we use that for line-height and spacing for a consistent look-and-feel across the document. You can read our [detailed analysis of "REMs vs EMs for spacing"](https://github.com/bbc/simorgh/blob/latest/docs/Spacing-Units.md) for more information.

## Contributing

When **adding** a new export to this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/gel-foundations/index.test.jsx#L13-L59) also need to be updated. When **removing** an exisiting export from this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/gel-foundations/index.test.jsx#L13-L59) need to be updated and the package version requires a major change (EG: 1.2.1 -> 2.0.0) as this would be considered a breaking change as functionality is being removed.

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/bbc/psammead/blob/latest/LICENSE).
