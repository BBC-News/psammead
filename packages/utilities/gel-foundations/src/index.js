import * as breakpoints from './breakpoints';
import * as propTypes from './prop-types';
import * as scripts from './scripts';
import * as spacings from './spacings';
import * as typography from './typography';

export { breakpoints };
export { propTypes };
export { scripts };
export { spacings };
export { typography };
