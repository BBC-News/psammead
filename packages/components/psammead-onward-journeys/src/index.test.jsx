import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { render } from '@testing-library/react';
import { latin } from '@bbc/gel-foundations/scripts';
import OnwardJourneys from '.';

const onwardJourneysData = [
  {
    name: 'Climate change',
    href: 'https://www.bbc.co.uk/news/topics/cmj34zmwm1zt/climate-change',

    url: 'https://www.bbc.co.uk/news/topics/cmj34zmwm1zt/climate-change',
    id: 'cmj34zmwm1zt',
    records: [
      {
        headline: "Climate change 'making mountaineering riskier'",
        url: 'https://www.bbc.co.uk/news/science-environment-50237551',
        image: {
          height: 549,
          width: 976,
          href:
            'http://c.files.bbci.co.uk/14A56/production/_109466548_crevasses.jpg',
          originCode: 'cpsprodpb',
          caption:
            'Crevasses are becoming difficult to cross in some mountainous regions',
          altText: 'Mountaineers climbing the Alps',
          copyrightHolder: 'Getty Images',
          allowSyndication: true,
        },
      },
      {
        headline: "Corbyn plans 'green industrial revolution'",
        url: 'https://www.bbc.co.uk/news/uk-politics-50318521',
        image: {
          height: 576,
          width: 1024,
          href:
            'http://c.files.bbci.co.uk/D615/production/_109550845_p07t5pvs.jpg',
          originCode: 'cpsprodpb',
          altText: 'Jeremy Corbyn',
          copyrightHolder: 'Reuters',
          allowSyndication: true,
        },
      },
      {
        headline: 'How would the Greens fund climate pledge?',
        url: 'https://www.bbc.co.uk/news/uk-politics-50315720',
        image: {
          height: 549,
          width: 976,
          href:
            'http://c.files.bbci.co.uk/AAAE/production/_109549634_gettyimages-976800330.jpg',
          originCode: 'cpsprodpb',
          altText: 'Wind turbines',
          copyrightHolder: 'Getty Images',
          allowSyndication: true,
        },
      },
      {
        headline: "Green Party: 'This must be the climate election'",
        url: 'https://www.bbc.co.uk/news/uk-politics-50317421',
        image: {
          height: 576,
          width: 1024,
          href:
            'http://c.files.bbci.co.uk/036A/production/_109547800_p07t5lh8.jpg',
          originCode: 'cpsprodpb',
          altText: 'Sian Berry, Carla Denyer and Amelia Womack',
          copyrightHolder: 'PA Media',
          allowSyndication: true,
        },
      },
      {
        headline: "Nuclear fusion: 'A question of when, not if'",
        url: 'https://www.bbc.co.uk/news/science-environment-50267017',
        image: {
          height: 261,
          width: 464,
          href:
            'http://c.files.bbci.co.uk/5494/production/_109525612_gettyimages-1095967942.jpg',
          originCode: 'cpsprodpb',
          caption: "An artist's impression of how a fusion reactor might look",
          altText: 'fusion',
          copyrightHolder: 'Getty Images',
          allowSyndication: true,
        },
      },
    ],
  },
  {
    name: 'Environment',
    url: 'https://www.bbc.co.uk/news/topics/cnx753jenyjt/environment',
    id: 'cnx753jenyjt',
    records: [
      {
        headline: 'How Shropshire ideas may help the rewilding debate',
        url: 'https://www.bbc.co.uk/news/uk-england-shropshire-50169027',
        image: {
          height: 549,
          width: 976,
          href:
            'http://c.files.bbci.co.uk/BF0A/production/_109360984_rewilding.jpg',
          originCode: 'cpsprodpb',
          caption:
            "In the Cambrian Mountains there's strong local opposition to rewilding",
          altText:
            "In the Cambrian Mountains there's strong local opposition to rewilding",
          copyrightHolder: 'BBC',
          allowSyndication: true,
        },
      },
      {
        headline: "Farmers' anger over slurry pollution rule changes",
        url: 'https://www.bbc.co.uk/news/uk-wales-50332308',
        image: {
          height: 549,
          width: 976,
          href:
            'http://c.files.bbci.co.uk/3ED6/production/_109568061_gettyimages-1148622395.jpg',
          originCode: 'cpsprodpb',
          altText: 'A cow sticking its tongue out',
          copyrightHolder: 'Getty Images',
          allowSyndication: true,
        },
      },
      {
        headline: 'This must be the climate election - Greens',
        url: 'https://www.bbc.co.uk/news/uk-politics-50305284',
        image: {
          height: 549,
          width: 976,
          href:
            'http://c.files.bbci.co.uk/15DF0/production/_109548598_mediaitem109548595.jpg',
          originCode: 'cpsprodpb',
          caption:
            'Co-leader Sian Berry, candidate Carla Denyer and deputy leader Amelia Womack',
          altText:
            'Sian Berry, Bristol West Candidate Carla Denyer and deputy leader Amelia Womack',
          copyrightHolder: 'PA Media',
          allowSyndication: true,
        },
      },
      {
        headline: "Plans to double city's green space",
        url: 'https://www.bbc.co.uk/news/uk-wales-50303472',
        image: {
          height: 549,
          width: 976,
          href:
            'http://c.files.bbci.co.uk/6468/production/_109540752_rivertawegreencorridor.png',
          originCode: 'cpsprodpb',
          altText: 'River Tawe Wildlife Corridor',
          copyrightHolder: 'Swansea Council',
          allowSyndication: false,
        },
      },
      {
        headline: "Do your 'plastic-free' teabags contain plastic?",
        url: 'https://www.bbc.co.uk/news/uk-50260687',
        image: {
          height: 549,
          width: 976,
          href:
            'http://c.files.bbci.co.uk/7ED8/production/_109427423_gettyimages-93195303.jpg',
          originCode: 'cpsprodpb',
          altText: 'Tea bag in cup',
          copyrightHolder: 'Getty Images',
          allowSyndication: true,
        },
      },
      {
        headline: 'How much will this be a climate change election?',
        url: 'https://www.bbc.co.uk/news/science-environment-50307304',
        image: {
          height: 549,
          width: 976,
          href:
            'http://c.files.bbci.co.uk/96D6/production/_109541683_mediaitem109541681.jpg',
          originCode: 'cpsprodpb',
          altText: 'Solar panels',
          copyrightHolder: 'PA Media',
          allowSyndication: true,
        },
      },
    ],
  },
];

describe('OnwardJourneys', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <OnwardJourneys
      onwardJourneysData={onwardJourneysData}
      service="news"
      script={latin}
    />,
  );

  it('should render component heading', () => {
    const { container } = render(
      <OnwardJourneys
        onwardJourneysData={onwardJourneysData}
        service="news"
        script={latin}
      />,
    );
    expect(container.querySelector('h2').textContent).toEqual(
      'More from the BBC',
    );
  });
});
