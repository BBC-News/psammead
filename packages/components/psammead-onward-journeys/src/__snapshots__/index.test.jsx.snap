// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnwardJourneys should render correctly 1`] = `
.c13 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c0 {
  padding-top: calc(4 * 1rem);
  padding-bottom: calc(2 * 1rem);
  background-color: #D5D0CD;
}

.c11 {
  padding: 0 1rem 0 1rem;
  min-width: 20%;
}

.c6 {
  padding: 1rem;
}

.c7 {
  border: none;
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #222222;
  background-color: #D5D0CD;
  padding: 0.5rem 1rem;
}

.c7:first-child {
  background-color: #222222;
  color: #FFFFFF;
}

.c2 {
  overflow: hidden;
  padding: 0;
  margin: 0;
  border: 0;
}

.c3 {
  position: absolute;
  top: 6.5rem;
  right: 20rem;
}

.c4 {
  border: none;
  background-color: #D5D0CD;
}

.c8 {
  display: block;
  list-style-type: none;
  margin: 0 1rem;
  padding: 0;
}

.c10 {
  background-color: #FFFFFF;
  display: inline-block;
  width: calc(20% - 4 * 0.5rem);
  margin: 0 0.5rem 1rem 0.5rem;
}

.c9 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12 {
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c14 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  margin: 1rem;
}

.c14:visited {
  color: #6E6E73;
  border-bottom: 1px solid #6E6E73;
}

.c14:focus,
.c14:hover {
  border-bottom: 2px solid #B80000;
  color: #B80000;
}

.c1 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  margin: 0 1rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c12 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c12 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c14 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c14 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    More from the BBC
  </h2>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <button
        class="c4"
        type="button"
      >
        <span
          class="c5"
        >
          Scroll carousel left
        </span>
        <svg
          enable-background="new 0 0 18 22"
          height="44px"
          version="1.1"
          viewBox="0 0 44 44"
          width="44px"
          x="0px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <g
            fill="#6e6e73"
            id="arrow-left"
          >
            <rect
              fill="#6e6e73"
              height="44"
              width="44"
            />
            <rect
              fill="#f2f2f2"
              height="40"
              width="40"
              x="2"
              y="2"
            />
            <polygon
              points="13,22 31,33 31,28.1 20.9,22 31,15.9 31,11"
            />
          </g>
        </svg>
      </button>
      <button
        class="c4"
        type="button"
      >
        <span
          class="c5"
        >
          Scroll carousel right
        </span>
        <svg
          enable-background="new 0 0 44 44"
          height="44px"
          version="1.1"
          viewBox="0 0 44 44"
          width="44px"
          x="0px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <g
            fill="#222222"
            id="arrow-right"
          >
            <rect
              fill="#222222"
              height="44"
              width="44"
            />
            <rect
              fill="#FFFFFF"
              height="40"
              width="40"
              x="2"
              y="2"
            />
            <polygon
              points="31,22 13,11 13,15.9 23.1,22 13,28.1 13,33"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="c6"
    >
      <button
        class="c7"
      >
        Climate change
      </button>
      <button
        class="c7"
      >
        Environment
      </button>
      <button
        class="c7"
      />
    </div>
  </div>
  <ul
    class="c8"
  >
    <a
      class="c9"
      href="https://www.bbc.co.uk/news/science-environment-50237551"
    >
      <li
        class="c10"
      >
        <div
          class="c11"
        >
          <h4
            class="c12"
          >
            Climate change 'making mountaineering riskier'
          </h4>
        </div>
        <img
          alt="Mountaineers climbing the Alps"
          class="c13"
          src="http://c.files.bbci.co.uk/14A56/production/_109466548_crevasses.jpg"
          width="100%"
        />
      </li>
    </a>
    <a
      class="c9"
      href="https://www.bbc.co.uk/news/uk-politics-50318521"
    >
      <li
        class="c10"
      >
        <div
          class="c11"
        >
          <h4
            class="c12"
          >
            Corbyn plans 'green industrial revolution'
          </h4>
        </div>
        <img
          alt="Jeremy Corbyn"
          class="c13"
          src="http://c.files.bbci.co.uk/D615/production/_109550845_p07t5pvs.jpg"
          width="100%"
        />
      </li>
    </a>
    <a
      class="c9"
      href="https://www.bbc.co.uk/news/uk-politics-50315720"
    >
      <li
        class="c10"
      >
        <div
          class="c11"
        >
          <h4
            class="c12"
          >
            How would the Greens fund climate pledge?
          </h4>
        </div>
        <img
          alt="Wind turbines"
          class="c13"
          src="http://c.files.bbci.co.uk/AAAE/production/_109549634_gettyimages-976800330.jpg"
          width="100%"
        />
      </li>
    </a>
    <a
      class="c9"
      href="https://www.bbc.co.uk/news/uk-politics-50317421"
    >
      <li
        class="c10"
      >
        <div
          class="c11"
        >
          <h4
            class="c12"
          >
            Green Party: 'This must be the climate election'
          </h4>
        </div>
        <img
          alt="Sian Berry, Carla Denyer and Amelia Womack"
          class="c13"
          src="http://c.files.bbci.co.uk/036A/production/_109547800_p07t5lh8.jpg"
          width="100%"
        />
      </li>
    </a>
    <a
      class="c9"
      href="https://www.bbc.co.uk/news/science-environment-50267017"
    >
      <li
        class="c10"
      >
        <div
          class="c11"
        >
          <h4
            class="c12"
          >
            Nuclear fusion: 'A question of when, not if'
          </h4>
        </div>
        <img
          alt="fusion"
          class="c13"
          src="http://c.files.bbci.co.uk/5494/production/_109525612_gettyimages-1095967942.jpg"
          width="100%"
        />
      </li>
    </a>
  </ul>
  <a
    class="c14"
    href="https://www.bbc.co.uk/news/topics/cmj34zmwm1zt/climate-change"
  >
    Climate change
  </a>
</div>
`;
